includes/models/model.llms.order.php

```diff
@@ -1318,36 +1318,25 @@ public function has_trial_ended() {
 	}
 
 	/**
-	 * Initialize a pending order
+	 * Initializes a new order with user, plan, gateway, and coupon metadata.
 	 *
-	 * Used during checkout.
 	 * Assumes all data passed in has already been validated.
 	 *
 	 * @since 3.8.0
 	 * @since 3.10.0 Unknown.
 	 * @since 5.3.0 Don't set unused legacy property `date_billing_end`.
+	 * @since [version] Use `LLMS_Order::set_user_data()` to update user data.
 	 *
-	 * @param LLMS_Student         $person  The LLMS_Student placing the order.
-	 * @param LLMS_Access_Plan     $plan    The purchase LLMS_Access_Plan.
-	 * @param LLMS_Payment_Gateway $gateway The LLMS_Payment_Gateway used.
-	 * @param LLMS_Coupon          $coupon  LLMS_Coupon if a coupon was used or false.
+	 * @param array|LLMS_Student|WP_User|integer $user_data User info for the person placing the order. See
```

Comment:
Passing in an *array* of information in favor of a fully registered user object is the crux of all of this... This allows us to save everything to the order so we can still create a pending order, but doesn't create the user (and possibly more importantly trigger user registration hooks which do things like send emails)

There are interface changes that will need to be made to accommodate an order that doesn't have a user ID.

Additionally, I have opted to *not* save custom field data or passwords to the order.

I considered doing this as a way to "resume" checkout later but the idea here is to complete everything in the same session. Based on how I'm building things (that I haven't pushed yet) it won't be necessary to reference this stuff from the order for user creation later as it'll still be available for future AJAX requests to confirm or complete the order.

Reply:
Ok for the reasoning "everything will be completed in the same session" but still, if there's a way to "resume", then custom fields should be saved to me. Why cannot they?

Reply:
We *can* store all the custom fields but in doing so it will require us to create a method for doing so which won't be strictly necessary given the way this is being built.

Essentially it feels like we'd be storing data for the sake of doing so instead of because it's required / necessary.

The real problem I have here too is with passwords... I know it's possible but we'll need to figure out how to properly hash/store the password into the order so then later we can pull it off the order and store it again (but without hashing it this time because it's already hashed). I'm sure this is possible but I haven't worked through how to do it. And I don't think this is essential for any of the existing payment gateways we're working with... Maybe I'm being short-sighted here but it feels like a lot of code that we're going to have to write in order to solve a problem that we don't currently have. I'm looking primarily at PayPal where there is a 2-step process.

The real concern would be a network interruption scenario. In this case the user is going to need to start on step one and re-enter their password.